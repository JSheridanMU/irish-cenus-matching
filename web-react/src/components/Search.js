import React, { useState, useEffect } from 'react'
import { useTheme } from '@material-ui/core/styles'
import {
  FormControl,
  FormLabel,
  Grid,
  Paper,
  RadioGroup,
  TextField,
  Radio,
  FormControlLabel,
  Select,
  InputLabel,
  MenuItem,
} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import Title from './Title'
import clsx from 'clsx'

import SearchResults from './SearchResults'

const initialValues = {
  year: '1901',
  forename: '',
  surname: '',
  county: '',
  townland: '',
  ded: '',
  age: '',
  sex: 'both',
}

export default function Search() {
  const theme = useTheme()

  const useStyles = makeStyles((theme) => ({
    root: {
      '& .MuiFormControl-root': {
        width: '80%',
        margin: theme.spacing(1),
      },
      display: 'flex',
    },
    paper: {
      padding: theme.spacing(2),
      display: 'flex',
      overflow: 'auto',
      flexDirection: 'column',
    },
    fixedHeight: {
      height: 450,
    },
  }))
  const classes = useStyles(theme)
  const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight)

  const [values, setValues] = useState(initialValues)

  const handleInputChange = (e) => {
    const { name, value } = e.target
    setValues({
      ...values,
      [name]: value,
    })
  }

  return (
    <React.Fragment>
      {console.log(values)}
      <Grid container spacing={4}>
        <Grid item xs={12} md={12} lg={12}>
          <Paper className={fixedHeightPaper}>
            <Title>Search</Title>
            <form className={classes.root}>
              <Grid container>
                <Grid item xs={6}>
                  <FormControl>
                    <FormLabel>Year</FormLabel>
                    <RadioGroup
                      row
                      name="year"
                      value={values.year}
                      onChange={handleInputChange}
                    >
                      <FormControlLabel
                        value="1901"
                        control={<Radio />}
                        label="1901"
                      />
                      <FormControlLabel
                        value="1911"
                        control={<Radio />}
                        label="1911"
                      />
                    </RadioGroup>
                  </FormControl>
                  <TextField
                    variant="outlined"
                    label="Surname"
                    name="surname"
                    value={values.surname}
                    onChange={handleInputChange}
                  />
                  <TextField
                    variant="outlined"
                    label="Forename"
                    name="forename"
                    value={values.forename}
                    onChange={handleInputChange}
                  />
                  <FormControl>
                    <FormLabel>Sex</FormLabel>
                    <RadioGroup
                      row
                      name="sex"
                      value={values.sex}
                      onChange={handleInputChange}
                    >
                      <FormControlLabel
                        value="male"
                        control={<Radio />}
                        label="Male"
                      />
                      <FormControlLabel
                        value="female"
                        control={<Radio />}
                        label="Female"
                      />
                      <FormControlLabel
                        value="both"
                        control={<Radio />}
                        label="Both"
                      />
                    </RadioGroup>
                  </FormControl>
                  <TextField
                    variant="outlined"
                    label="Age"
                    name="age"
                    type="number"
                    value={values.age}
                    onChange={handleInputChange}
                  />
                </Grid>
                <Grid item xs={6}>
                  <FormControl variant="outlined">
                    <InputLabel>County</InputLabel>
                    <Select
                      value={values.county}
                      onChange={handleInputChange}
                      label="County"
                      name="county"
                    >
                      <MenuItem value="">
                        <em>None</em>
                      </MenuItem>
                      <MenuItem value={'Carlow'}>Carlow</MenuItem>
                      <MenuItem value={'Dublin'}>Dublin</MenuItem>
                      <MenuItem value={'Wexford'}>Wexford</MenuItem>
                    </Select>
                  </FormControl>
                  <TextField
                    variant="outlined"
                    label="Townland/Street"
                    name="townland"
                    value={values.townland}
                    onChange={handleInputChange}
                  />
                  <TextField
                    variant="outlined"
                    label="DED"
                    name="ded"
                    value={values.ded}
                    onChange={handleInputChange}
                  />
                </Grid>
              </Grid>
            </form>
          </Paper>
        </Grid>
      </Grid>
      <SearchResults></SearchResults>
    </React.Fragment>
  )
}
