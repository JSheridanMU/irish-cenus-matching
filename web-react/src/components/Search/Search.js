import React, { useState } from 'react'
import { useTheme } from '@material-ui/core/styles'
import { Grid, Paper } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import Title from '../Title/Title'
import clsx from 'clsx'
import SearchResults from '../SearchResults/SearchResults'
import { UseSearch, Form } from '../UseSearch'
import FieldImporter from '../form-fields/FieldImporter'
import * as DataService from '../../services/DataService'

const initialValues = {
  year: '1911',
  forename: '',
  surname: '',
  county: '',
  townland: '',
  ded: '',
  age: '',
  sex: 'both',
}

export default function Search() {
  const theme = useTheme()
  const [search, setSearch] = useState(false)

  const useStyles = makeStyles((theme) => ({
    root: {
      display: 'flex',
    },
    paper: {
      padding: theme.spacing(2),
      display: 'flex',
      overflow: 'auto',
      flexDirection: 'column',
    },
    fixedHeight: {
      height: 450,
    },
  }))
  const classes = useStyles(theme)
  const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight)

  const { values, handleInputChange, resetForm } = UseSearch(initialValues)

  const handleSubmit = () => {
    setSearch(true)
  }

  return (
    <React.Fragment>
      <Grid container spacing={4}>
        <Grid item xs={12} md={12} lg={12}>
          <Paper className={fixedHeightPaper}>
            <Title>Search</Title>
            <Form>
              <Grid container>
                <Grid item xs={6}>
                  <FieldImporter.Input
                    label="Forename"
                    name="forename"
                    value={values.forename}
                    onChange={handleInputChange}
                  />
                  <FieldImporter.Input
                    name="surname"
                    label="Surname"
                    value={values.surname}
                    onChange={handleInputChange}
                  />
                  <FieldImporter.Input
                    label="Age"
                    name="age"
                    value={values.age}
                    onChange={handleInputChange}
                    type="number"
                  />
                  <FieldImporter.RadioGroup
                    label="Sex"
                    name="sex"
                    value={values.sex}
                    onChange={handleInputChange}
                    items={DataService.getSexes()}
                  />
                </Grid>
                <Grid item xs={6}>
                  <FieldImporter.Select
                    name="county"
                    label="County"
                    value={values.county}
                    onChange={handleInputChange}
                    options={DataService.getCounties()}
                  />
                  <FieldImporter.Input
                    label="Townland/Street"
                    name="townland"
                    value={values.townland}
                    onChange={handleInputChange}
                  />
                  <FieldImporter.Input
                    label="DED"
                    name="ded"
                    value={values.ded}
                    onChange={handleInputChange}
                  />
                  <FieldImporter.RadioGroup
                    label="Year"
                    name="year"
                    value={values.year}
                    onChange={handleInputChange}
                    items={DataService.getYears()}
                  />
                  <div>
                    <FieldImporter.Button
                      text="Submit"
                      onClick={handleSubmit}
                    />
                    <FieldImporter.Button
                      text="Reset"
                      color="default"
                      onClick={resetForm}
                    />
                  </div>
                </Grid>
              </Grid>
            </Form>
          </Paper>
        </Grid>
      </Grid>
      {search ? <SearchResults values={values} /> : null}
    </React.Fragment>
  )
}
