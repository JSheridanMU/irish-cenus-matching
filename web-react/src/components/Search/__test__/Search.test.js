import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client'
import Search from '.././Search'

afterEach(cleanup)

const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_URI,
  cache: new InMemoryCache(),
})

test('renders without crashing', () => {
  render(<Search />)
})

test('renders with search fields', () => {
  const { getByText, getAllByText } = render(<Search />)

  getByText('Search')
  getByText('Sex')
  getByText('Year')
  getByText('Male')
  getByText('Female')
  getByText('Both')
  getByText('1901')
  getByText('1911')
  getAllByText('Forename')
  getAllByText('Surname')
  getAllByText('Age')
  getAllByText('County')
  getAllByText('Townland/Street')
  getAllByText('DED')
})

test('submission triggers search', () => {
  const { getByText, getByTestId } = render(
    <ApolloProvider client={client}>
      <Search />
    </ApolloProvider>
  )

  const button = getByText('Submit')
  fireEvent.click(button)
  getByTestId('search rendered')
})

test('matches snapshot', () => {
  const tree = renderer.create(<Search />).toJSON()
  expect(tree).toMatchSnapshot()
})
