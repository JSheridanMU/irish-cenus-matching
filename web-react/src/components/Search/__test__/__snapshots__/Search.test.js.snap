// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
  >
    <div
      className="MuiPaper-root makeStyles-paper-60 makeStyles-fixedHeight-61 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        Search
      </h2>
      <form
        autoComplete="off"
        className="makeStyles-root-62"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Forename
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="forename"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      Forename
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Surname
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="surname"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      Surname
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Age
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="age"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      Age
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root"
            >
              <label
                className="MuiFormLabel-root"
              >
                Sex
              </label>
              <div
                className="MuiFormGroup-root MuiFormGroup-row"
                role="radiogroup"
              >
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-67 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-70"
                        disabled={false}
                        name="sex"
                        onChange={[Function]}
                        type="radio"
                        value="M"
                      />
                      <div
                        className="PrivateRadioButtonIcon-root-71"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-72"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Male
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-67 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-70"
                        disabled={false}
                        name="sex"
                        onChange={[Function]}
                        type="radio"
                        value="F"
                      />
                      <div
                        className="PrivateRadioButtonIcon-root-71"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-72"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Female
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-67 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-68 Mui-checked MuiIconButton-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={true}
                        className="PrivateSwitchBase-input-70"
                        disabled={false}
                        name="sex"
                        onChange={[Function]}
                        type="radio"
                        value="both"
                      />
                      <div
                        className="PrivateRadioButtonIcon-root-71 PrivateRadioButtonIcon-checked-73"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-72"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Both
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              className="MuiFormControl-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                County
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-county"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-county"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </div>
                <input
                  name="county"
                  onAnimationStart={[Function]}
                  type="hidden"
                  value=""
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      County
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Townland/Street
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="townland"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      Townland/Street
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                DED
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="ded"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-65"
                  >
                    <span>
                      DED
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root"
            >
              <label
                className="MuiFormLabel-root"
              >
                Year
              </label>
              <div
                className="MuiFormGroup-root MuiFormGroup-row"
                role="radiogroup"
              >
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-67 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-70"
                        disabled={false}
                        name="year"
                        onChange={[Function]}
                        type="radio"
                        value="1901"
                      />
                      <div
                        className="PrivateRadioButtonIcon-root-71"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-72"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    1901
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-67 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-68 Mui-checked MuiIconButton-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={true}
                        className="PrivateSwitchBase-input-70"
                        disabled={false}
                        name="year"
                        onChange={[Function]}
                        type="radio"
                        value="1911"
                      />
                      <div
                        className="PrivateRadioButtonIcon-root-71 PrivateRadioButtonIcon-checked-73"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-72"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    1911
                  </span>
                </label>
              </div>
            </div>
            <div>
              <button
                className="MuiButtonBase-root MuiButton-root makeStyles-root-74 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label makeStyles-label-75"
                >
                  Submit
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root makeStyles-root-74 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label makeStyles-label-75"
                >
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
